src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @see http://www.php-fig.org/psr/psr-3/
+ */
+final class Logger extends AbstractLogger
+{
+    /**
+     * @var array
+     */
+    private static $defaultLevelToStream = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stdout',
+        LogLevel::NOTICE => 'php://stdout',
+        LogLevel::INFO => 'php://stdout',
+        LogLevel::DEBUG => false,
+    );
+
+    private $levelToStream;
+    private $formatPattern;
+
+    /**
+     * @var bool
+     */
+    private $errored = false;
+
+    /**
+     * @param array  $levelToStream
+     * @param string $formatPattern
+     */
+    public function __construct(array $levelToStream = array(), $formatPattern = '[%s] %s')
+    {
+        $this->levelToStream = $levelToStream + self::$defaultLevelToStream;
+        $this->formatPattern = $formatPattern;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset($this->levelToStream[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (!$this->errored) {
```

Comment:
in fact, I'm skeptical about the need to write conditionally to stderr vs stdout.
despite its name, stderr is not only for errors but is for also for diagnostics, which all logs are to me,
especially in the simple logger we target.